### PR TITLE
Combines legacy_population and sanitizes death queries

### PR DESCRIPTION
**Pull request by Jordie0608**
Currently `playercount` and `admincount` are inserted into the DB via two queries run at the same time; I have absolutely no idea why this is the case, so the two queries have been combined into one.

Additionally `sql_report_death()` has had proper sanitization added to all its values partly because sanitization is good but mostly because Somepotato keeps mentioning it. When I get home I'll also add to it the damage types that aren't recorded by the table for what is also probably not a good reason.
DNM @MrStonedOne for table change (gimme db already ya nerd)
@nfreader

### CODE REVIEW

**Comment by Jordie0608** [comment]
Not sure why Git thinks the paths in `server_maintenance.dm` are being changed to what they already are in master; I'll check it out later.

**Comment by Jordie0608** [comment]
Oh well, didn't figure out why and am not bothered enough by it to try and resolve; if in doubt here's a comparison of the two 
capture

**Comment by nfreader** [comment]
👍 Yay more data!  

👎 No server being recorded
👎 Only way to match up populations is by matching `round_end` on `feedback_feedback` and I am cri

**Comment by Jordie0608** [comment]
>No server being recorded

The IP and port is tracked on `death` but not on `legacy_population` (although I could add it you want) since...
> Only way to match up populations is by matching `round_end` on `feedback_feedback` and I am cri

... as far as I can tell the table was designed solely for averaging the probably combined populations of the servers recorded every 10 minutes without regard for the rounds. Tracking the per-round population might be better handled from within `feedback` or by the elusive stats refactor.

**Comment by MrStonedOne** [comment]
it does need a server/port

**Comment by Jordie0608** [comment]
Well there's no reason not too.

**Comment by Jordie0608** [comment]
@MrStonedOne Rather than changing the columns to be nullable I've decided to just parse a null IP or CID as 0 before the insert query because:
- The query for bans checks `OR` on IP and CID, unless I misunderstood a significant part of my networking classes a connecting client's IP will never be 0, I assume the same for CID, so it'll effectively be the same as if the check were for `null`
- Because of the optimisation queries all the empty table data was already set to 0
- It keeps consistency with other IP fields, sans `unbanned_ip` I guess and I like that
Unless there's some interaction I've overlooked I'm pretty sure this should work the same.

**Comment by MrStonedOne** [comment]
merge at will, just nag me when you do and i'll change the db.

File: code/orphaned_procs/statistics.dm [hunk 1]
```diff
@@ -206,68 +206,47 @@ var/datum/feedback/blackbox = new()
...
-	var/sqlpod = sanitizeSQL(podname)
-	var/sqlspecial = sanitizeSQL(L.mind.special_role)
+	var/sqlkey = sanitizeSQL(L.ckey)
 	var/sqljob = sanitizeSQL(L.mind.assigned_role)
+	var/sqlspecial = sanitizeSQL(L.mind.special_role)
+	var/sqlpod = sanitizeSQL(placeofdeath.name)
 	var/laname
 	var/lakey
 	if(L.lastattacker)
 		laname = sanitizeSQL(L.lastattacker:real_name)
```
Comment (AnturK) [inline code/orphaned_procs/statistics.dm]: Ew, colons.

Comment (Jordie0608) [inline code/orphaned_procs/statistics.dm]: Fiiiiine

**Comment by AnturK** [comment]
@MrStonedOne

File: SQL/database_changelog.txt [hunk 2]
```diff
@@ -1,3 +1,13 @@
+10 March 2017, by Jordie0608
+
+Modified table 'death', adding the columns 'toxloss', 'cloneloss', and 'staminaloss' and table 'legacy_population', adding the columns 'server_ip' and 'server_port'.
+
+ALTER TABLE `feedback`.`death` ADD COLUMN `toxloss` SMALLINT(5) UNSIGNED NOT NULL AFTER `oxyloss`, ADD COLUMN `cloneloss` SMALLINT(5) UNSIGNED NOT NULL AFTER `toxloss`, ADD COLUMN `staminaloss` SMALLINT(5) UNSIGNED NOT NULL AFTER `cloneloss`;
```
Comment (MrStonedOne) [inline SQL/database_changelog.txt]: oh hey, you forgot the legacy_pop table lines

Comment (Jordie0608) [inline SQL/database_changelog.txt]: https://github.com/tgstation/tgstation/commit/cdce692c32baeb1e090f1945243036b018396894

**Comment by nfreader** [comment]
Woo!

**Pull request merged**